sqlmesh/core/loader.py

```diff
@@ -208,32 +219,41 @@ def _load_external_models(
         if external_models_path.exists() and external_models_path.is_dir():
             paths_to_load.extend(self._glob_paths(external_models_path, extension=".yaml"))
 
+        def _load() -> t.List[Model]:
+            try:
+                with open(path, "r", encoding="utf-8") as file:
+                    return [
+                        create_external_model(
+                            defaults=self.config.model_defaults.dict(),
+                            path=path,
+                            project=self.config.project,
+                            audit_definitions=audits,
+                            **{
+                                "dialect": self.config.model_defaults.dialect,
+                                "default_catalog": self.context.default_catalog,
+                                **row,
+                            },
+                        )
+                        for row in YAML().load(file.read())
+                    ]
+            except Exception as ex:
+                raise ConfigError(f"Failed to load model definition at '{path}'.\n{ex}")
+
+        cache = self._cache()
```

Comment:
I don't understand how this works. Doesn't dbt's cache require these arguments?
```
                project=project,
                macros_max_mtime=macros_max_mtime,
                yaml_max_mtimes=yaml_max_mtimes,
                ```

Reply:
you're right, i need to change the api one more time